src/Trinity.C/src/Storage/MemoryTrunk/MemoryTrunk.Memory.cpp

```diff
@@ -250,8 +263,7 @@ namespace Storage
         size_t append_head = cp - buffer;
         if (Memory::MemoryCommit(trunkPtr, append_head) == NULL)
         {
-            free(buffer);
-            return -1;
+            Trinity::Diagnostics::FatalError("ReloadImpl: MemoryTrunk {0} fails to reclaim memory after data's been moved to a temporary buffer.", TrunkId);
```

Comment:
Should we recover the trunk data? because once it panics and exits, the data is lost.